cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/config/EnvConfigProvider.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2021 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+package com.linkedin.kafka.cruisecontrol.metricsreporter.config;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.kafka.common.config.ConfigData;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.common.config.provider.ConfigProvider;
+
+/**
+ * Enables config properties of the form ${env:ENV_KEY}
+ */
+public class EnvConfigProvider implements ConfigProvider {
```

Comment:
Should we use this across the project (in `cruise-control` module) to avoid functionality duplication?
@efeg would you please comment on this? I've seen a couple of examples of using metrics reporter classes in the `cruise-control` module (mainly the `metrics` and `exception` packages) but I'm not sure if this module is intended to be used as a common utilility-like module.

Reply:
Given the existing dependency, I agree that using this in `cruise-control` module is better than introducing functionality duplication.

That being said, the ideal solution, as you pointed out, is to have a utility-like module for such common Kafka-related logic, and use it from (1) `cruise-control` and (2) `cruise-control-metrics-reporter` modules.

I am ok with adopting the earlier approach and leaving the ideal solution to a future PR. Or if we would decide to go with the ideal solution and cleanup the dependency between `cruise-control` and `cruise-control-metrics-reporter` that is even better :)